### PR TITLE
got rid of duplicate modals

### PR DESCRIPTION
**Pull request by numbap**
### Description

List of proposed changes:

- Altered  SessionTimeout to use embedded modal component instead of creating a new modal instance. 
-

### What to test for/How to test
 - Click on link that would normally trigger beta modal. 
 - Wait 30 seconds for timer modal to appear.

### CODE REVIEW

File: pages/my-dashboard.js [hunk 1]
```diff
@@ -73,7 +88,7 @@ export default function MyDashboard(props) {
               <MostReqTasks
                 taskListMR={mostReq}
                 dataCy="most-requested"
-                openModal={openModal}
+                openModal={(link) => openModal('/', betaModal)}
```
Comment (nibivi77) [inline pages/my-dashboard.js:91]: `link` is never used.

File: pages/my-dashboard.js [hunk 2]
```diff
@@ -99,29 +114,21 @@ export default function MyDashboard(props) {
...
+      <SessionTimeout
+        popupStaySignedIn={props.popupStaySignedIn}
+        aaPrefix={props.aaPrefix}
+        openModal={(context) => openModal('/', context)}
+        closeModal={closeModal}
+      />
       <Modal
         className="flex justify-center bg-black/75 h-full"
-        isOpen={openModalWithLink.isOpen}
+        // isOpen={openModalWithLink.isOpen}
```
Comment (nibivi77) [inline pages/my-dashboard.js:125]: Should we keep this comment?

**Review by nibivi77 — APPROVED** [review]
Left a few comments. This is good to me though!

File: pages/my-dashboard.js [hunk 3]
```diff
@@ -89,7 +104,7 @@ export default function MyDashboard(props) {
                       acronym={acronym(card.title)}
                       taskList={taskList}
                       dataCy="task-group-list"
-                      openModal={openModal}
+                      openModal={(link) => openModal('/', betaModal)}
```
Comment (krischarbonneau) [inline pages/my-dashboard.js:107]: Same as Ben's first comment, `link` is never used here

Comment (numbap) [reply pages/my-dashboard.js]: I've completely rewritten the code.

File: pages/profile.js [hunk 4]
```diff
@@ -7,11 +7,12 @@ import { getProfileContent } from '../graphql/mappers/profile'
 import { getBetaBannerContent } from '../graphql/mappers/beta-banner-opt-out'
 import { getBetaPopupExitContent } from '../graphql/mappers/beta-popup-exit'
 import { getBetaPopupNotAvailableContent } from '../graphql/mappers/beta-popup-page-not-available'
+import { getAuthModalsContent } from '../graphql/mappers/auth-modals'
 import logger from '../lib/logger'
 import ProfileTasks from './../components/ProfileTasks'
 import Modal from 'react-modal'
 import React from 'react'
-import ExitBetaModal from '../components/ExitBetaModal'
+import ExitBeta from '../components/ExitBeta'
```
Comment (krischarbonneau) [inline pages/profile.js:15]: unused import, same with line 13

File: components/Layout.js [hunk 5]
```diff
@@ -119,10 +145,24 @@ export default function Layout(props) {
         {display.fullscreen ? (
           props.children
         ) : (
-          <LayoutContainer>{props.children}</LayoutContainer>
+          <LayoutContainer>
+            {cloneElement(props.children, { openModal, closeModal })}
```
Comment (krischarbonneau) [inline components/Layout.js:149]: I'm not huge on using this, given React's own documentation warns against using it, but it does work for the time being. I would prefer we look into a different option but that can be done separately as I fear this PR is large enough at this point

**Pull request merged**